invenio_rdm_records/services/errors.py

```diff
@@ -154,9 +153,9 @@ def __init__(self, record_id, community_id):
     def description(self):
         """Exception description."""
         return _(
-            "The record {record_id} in not included in the community {community_id}.".format(
-                record_id=self.record_id, community_id=self.community_id
-            )
+            "The record %(rec_id)s in not included in the community %(com_id)s.",
```

Comment:
ping @zzacharo @ntarocco @slint just flagging this to have your opinion

I find `%` operator for string formatting very unreadable and less flexible than `.format`, as far as I know there is negligible to no difference in performance during the evaluation

Reply:
i completely agree, the `%` format is not as nice as f-strings by any means!
however, from my understanding, they're required by `lazy_gettext()`...

Reply:
isn't lazy_gettext ok also with `.format(` ?
I remember f-strings not working with lazy_gettext but .format should be fine (to be verified - don't trust my memory!)

Reply:
Above is a correct analysis.

Correct usage:
```
lazy_gettext("String %(var)s", var="a var")
```

Incorrect usage:
```
lazy_gettext("String {var}".format(var="test")) # will try to translate "String "test" but only "String {var}" was extracted.
lazy_gettext("String {var}").format(var="test") # reduces to gettext and translates immediately.
```

Technically, ``gettext("String {var}").format(var="test")`` will yield a correct result, but you rarely know when you are working with lazy_gettext vs gettext, hence for consistency I would never use it.

Reply:
Btw, https://inveniordm.docs.cern.ch/develop/howtos/i18n/ is wrong in this case and I see we're missing a section as well https://inveniordm.docs.cern.ch/develop/howtos/i18n/#lazy-vs-non-lazy

Reply:
@lnielsen is not only the fastest merger in the west, but also beat me to this :smile: 

---

I just verified the results in a local instance, having the following code:
`invenio_rdm_records/requests/user_moderation/actions.py`
```python
    with force_locale("de_DE"):
        print(get_locale(), _("User %(user_id)s was blocked", user_id=user_id))
```
`invenio_rdm_records/translations/de/LC_MESSAGES/messages.po` (same for `de_DE` instead of `de`, and similar (with an empty `msgstr` in `invenio_rdm_records/translations/messages.pot`)
```
#: invenio_rdm_records/requests/user_moderation/actions.py:51
#, python-format
msgid "User %(user_id)s was blocked"
msgstr "Benutzer %(user_id)s wurde blockiert"
```
`invenio.cfg`
```python
I18N_LANGUAGES = [
    ('de', _('German')),
    ('de_DE', _('German')),
]
```

This will print the expected values:
```
[2023-09-11 13:58:01,421: WARNING/ForkPoolWorker-17] de_DE
[2023-09-11 13:58:01,421: WARNING/ForkPoolWorker-17]
[2023-09-11 13:58:01,432: WARNING/ForkPoolWorker-17] Benutzer 5 wurde blockiert
```

---

If I simply replace the `%(user_id)s` with `{user_id}` (f-string / format syntax) like so (in the code as well as the translation files):
```python
    with force_locale("de_DE"):
        print(get_locale(), _("User {user_id} was blocked", user_id=user_id))
```

Then I get the following logs:
```
[2023-09-11 14:01:09,448: WARNING/ForkPoolWorker-17] de_DE
[2023-09-11 14:01:09,449: WARNING/ForkPoolWorker-17]
[2023-09-11 14:01:09,460: WARNING/ForkPoolWorker-17] Benutzer {user_id} wurde blockiert
```
Which is also expected, since `Flask-Babel` uses `%` formatting and the syntax is different for that.
The text is translated correctly, but the `user_id=user_id` value is simply discarded.

---

Calling the `.format()` "inside" the translation:
```python
    with force_locale("de_DE"):
        print(get_locale(), _("User {user_id} was blocked".format(user_id=user_id)))
```

Will break the translation:
```
[2023-09-11 14:02:39,452: WARNING/ForkPoolWorker-17] de_DE
[2023-09-11 14:02:39,452: WARNING/ForkPoolWorker-17]
[2023-09-11 14:02:39,464: WARNING/ForkPoolWorker-17] User 5 was blocked
```

---

Lastly, calling it "outside" will work like `gettext`, but waste a few CPU cycles (due to the construction and evaluation of the `LazyString`):
```python
    with force_locale("de_DE"):
        print(get_locale(), _("User {user_id} was blocked").format(user_id=user_id))
```
```
[2023-09-11 14:06:14,551: WARNING/ForkPoolWorker-17] de_DE
[2023-09-11 14:06:14,551: WARNING/ForkPoolWorker-17]
[2023-09-11 14:06:14,551: WARNING/ForkPoolWorker-17] Benutzer 5 wurde blockiert
```